app/controllers/text_components_controller.rb

```diff
@@ -82,14 +82,16 @@ def set_form_data
     end
 
     def set_index_data
-      @text_components = TextComponent.includes(:triggers, :question_answers, :channels)
+      @diary_entry = DiaryEntry.new(report: @report, release: :final, moment: Time.zone.now)
+
+      @text_components = TextComponent.includes(:report, :channels, :triggers)
 
       filter_text_components
 
       @trigger_groups = @text_components.order('from_day, from_hour').group_by(&:trigger_ids)
 
       @trigger_groups = @trigger_groups.map do |trigger_ids, components|
-        [Trigger.find(trigger_ids), components]
+        [components.first.triggers, components]
```

Comment:
I don't get that one

Reply:
I’m not sure if this actually has any advantages, as db requests are automatically cached by active record, still I thought simply using the triggers already loaded might be better than finding them again by id.

Reply:
hm, I'm not quite sure if this is the same thing - but actually, yeah